cmd/meroxa/root/apps/deploy.go

```diff
@@ -524,6 +526,69 @@ func (d *Deploy) validate(ctx context.Context) error {
 	return turbineCLI.ValidateBranch(ctx, d.logger, d.path)
 }
 
+func (d *Deploy) checkResourceAvailability(ctx context.Context) error {
+	d.logger.Infof(ctx, "Checking resource availability for application %q (%s) before deployment...", d.appName, d.lang)
+
+	var resourceNames []string
+	var err error
+
+	switch d.lang {
+	case GoLang:
+		resourceNames, err = turbineGo.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case JavaScript, NodeJs:
+		resourceNames, err = turbineJS.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	case Python:
+		resourceNames, err = turbinePY.GetResourceNames(ctx, d.logger, d.path, d.appName)
+	default:
+		return fmt.Errorf("language %q not supported. %s", d.lang, LanguageNotSupportedError)
+	}
+
+	if err != nil {
+		return fmt.Errorf("unable to read resource definition from app: %s", err.Error())
+	}
+
+	switch {
+	case resourceNames == nil, len(resourceNames) == 0:
+		return errors.New("no resources defined in your Turbine app")
+	case len(resourceNames) < requiredNumberOfResources:
+		return fmt.Errorf(
+			"insufficient number of resources defined: %v."+
+				"Please add at least one source and one destination resource to your app definition",
+			resourceNames,
+		)
+	}
```

Comment:
I don't think we should include this check. I have a Turbine app that uses a Postgres source but uses only a function to push data into Salesforce (i.e. no destination).